### ISSUE TITLE
Typescript error "TS2307: Cannot find module '../aurelia.json'" within build tasks

### ISSUE THREAD

**Issue by suamikim** [aurelia/cli#493]
**I'm submitting a bug report** (i guess)

* **Library Version:**
1.0.8

**Please tell us about your environment:**
* **Operating System:**
Windows 10

* **Node Version:**
6.9.4

* **NPM Version:**
4.1.1

* **Browser:**
Not Browser related

* **Language:**
TypeScript 2.1.5

* **Current behavior:**
The line `import * as project from '../aurelia.json';` inside the build tasks (`aurelia_project\tasks\build.ts`, `aurelia_project\tasks\process-css.ts` etc.) reports the TypeScript error `TS2307: Cannot find module '../aurelia.json'` even though the file `aurelia.json` exists in the folder `aurelia_project`.

* **What is the expected behavior?**
Whilst building etc. is actually working as expected, I find it annoying & misleading to see TypeScript errors in my IDE (files & folders are marked red, the TypeScript panel shows those erros etc.) and I guess it will bite me when starting create a deploy process.

This can be reproduced by creating a new project using `au new` with the following options (CLI version ``):
* 3 Custom
* 2 TypeScript
* 3 Max minification
* 3 Sass
* 1 Yes (Unit Testing)
* 4 WebStorm
* 1 Yes (Create Project)
* 1 Yes (Install Dependencies)

The errors are reported within `PhpStorm 2016.3.2` as well as `VSCode 1.8.1 & 1.9.1`

**Ps.:** I'm not sure if this is related to https://github.com/aurelia/cli/issues/300 even though the title suggests otherwise since I believe that the issue there is about including JSON files within the source code, not within the build...

**Comment by sbolel** [comment]
Following https://github.com/aurelia/cli/issues/300#issuecomment-251072000, can you adding a file "`json-loader.d.ts`" in `custom_typings/` with the contents:

```ts
declare module "*.json" {
    const value: any;
    export default value;
}
```

**Comment by suamikim** [comment]
This resolves `TS2307: Cannot find module '../aurelia.json'` but leads to errors `TS2339:Property 'xxx' does not exist on type 'typeof "*.json"'.` when accessing the exported JSON:

E.g. in `process-css.ts`:

```
// ...
import * as project from '../aurelia.json';
// ...
export default function processCSS() {
	// ...
	
	return gulp.src(project.cssProcessor.source)
		// ...
};
```

The code `project.cssProcessor.source` leads to `TS2339:Property 'markupProcessor' does not exist on type 'typeof "*.json"'.`

**Comment by sbolel** [comment]
What if you change the import statement to:

```ts
import project from '../aurelia.json';
```

**Comment by suamikim** [comment]
Perfect, that did the trick!

**Comment by suamikim** [comment]
Unfortunately I was a little over-hasty since this prevents the CLI from building. E.g. `au run --watch` results in the following error:

```
Starting 'readProjectConfiguration'...
{ uid: 3,
  name: 'readProjectConfiguration',
  branch: false,
  error:
   { TypeError: Cannot read property 'build' of undefined
       at new <anonymous> (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:23:59)
       at Function.create (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:47:19)
       at Object.exports.src (project_dir\node_modules\aurelia-cli\lib\build\index.js:18:18)
       at readProjectConfiguration (project_dir\aurelia_project\tasks\build.ts:12:32)
       at bound (domain.js:280:14)
       at runBound (domain.js:293:12)
       at asyncRunner (project_dir\node_modules\async-done\index.js:36:18)
       at _combinedTickCallback (internal/process/next_tick.js:67:7)
       at process._tickDomainCallback (internal/process/next_tick.js:122:9)
     domain:
      Domain {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        members: [] },
     domainThrown: true },
  duration: [ 0, 4395008 ],
  time: 1487846456354 }
{ uid: 2,
  name: '<series>',
  branch: true,
  error:
   { TypeError: Cannot read property 'build' of undefined
       at new <anonymous> (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:23:59)
       at Function.create (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:47:19)
       at Object.exports.src (project_dir\node_modules\aurelia-cli\lib\build\index.js:18:18)
       at readProjectConfiguration (project_dir\aurelia_project\tasks\build.ts:12:32)
       at bound (domain.js:280:14)
       at runBound (domain.js:293:12)
       at asyncRunner (project_dir\node_modules\async-done\index.js:36:18)
       at _combinedTickCallback (internal/process/next_tick.js:67:7)
       at process._tickDomainCallback (internal/process/next_tick.js:122:9)
     domain:
      Domain {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        members: [] },
     domainThrown: true },
  duration: [ 0, 8992484 ],
  time: 1487846456358 }
{ uid: 1,
  name: '<series>',
  branch: true,
  error:
   { TypeError: Cannot read property 'build' of undefined
       at new <anonymous> (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:23:59)
       at Function.create (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:47:19)
       at Object.exports.src (project_dir\node_modules\aurelia-cli\lib\build\index.js:18:18)
       at readProjectConfiguration (project_dir\aurelia_project\tasks\build.ts:12:32)
       at bound (domain.js:280:14)
       at runBound (domain.js:293:12)
       at asyncRunner (project_dir\node_modules\async-done\index.js:36:18)
       at _combinedTickCallback (internal/process/next_tick.js:67:7)
       at process._tickDomainCallback (internal/process/next_tick.js:122:9)
     domain:
      Domain {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        members: [] },
     domainThrown: true },
  duration: [ 0, 10406481 ],
  time: 1487846456359 }
{ uid: 0,
  name: '<series>',
  branch: true,
  error:
   { TypeError: Cannot read property 'build' of undefined
       at new <anonymous> (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:23:59)
       at Function.create (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:47:19)
       at Object.exports.src (project_dir\node_modules\aurelia-cli\lib\build\index.js:18:18)
       at readProjectConfiguration (project_dir\aurelia_project\tasks\build.ts:12:32)
       at bound (domain.js:280:14)
       at runBound (domain.js:293:12)
       at asyncRunner (project_dir\node_modules\async-done\index.js:36:18)
       at _combinedTickCallback (internal/process/next_tick.js:67:7)
       at process._tickDomainCallback (internal/process/next_tick.js:122:9)
     domain:
      Domain {
        domain: null,
        _events: {},
        _eventsCount: 0,
        _maxListeners: undefined,
        members: [] },
     domainThrown: true },
  duration: [ 0, 12318247 ],
  time: 1487846456360 }
{ TypeError: Cannot read property 'build' of undefined
    at new <anonymous> (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:23:59)
    at Function.create (project_dir\node_modules\aurelia-cli\lib\build\bundler.js:47:19)
    at Object.exports.src (project_dir\node_modules\aurelia-cli\lib\build\index.js:18:18)
    at readProjectConfiguration (project_dir\aurelia_project\tasks\build.ts:12:32)
    at bound (domain.js:280:14)
    at runBound (domain.js:293:12)
    at asyncRunner (project_dir\node_modules\async-done\index.js:36:18)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickDomainCallback (internal/process/next_tick.js:122:9)
  domain:
   Domain {
     domain: null,
     _events: {},
     _eventsCount: 0,
     _maxListeners: undefined,
     members: [] },
  domainThrown: true }
```

**Comment by suamikim** [comment]
Now build is also working!
Thanks again!

Global `tsc -v` reports `Version 2.2.1` but the project is using the locally installed version `2.1.5` if this is still of any interest to you...

**Comment by suamikim** [comment]
Now build is also working!
Thanks again!

Global `tsc -v` reports `Version 2.2.1` but the project is using the locally installed version `2.1.5` if this is still of any interest to you...

**Comment by sbolel** [comment]
Awesome. Happy to hear it worked @suamikim. 

CC'ing @EisenbergEffect for visibility -- I'll create a PR for this if needed. What do you think?

**Comment by EisenbergEffect** [comment]
@sbolel Is there some way this can be achieved with out using `require`, but by using `import` instead?

**Comment by sbolel** [comment]
@EisenbergEffect 
- Not sure which of my 2 suggestions above fixed it for @suamikim. 
- I'll go back to a fresh clone to re-try, and will get back to you tomorrow.

**Comment by suamikim** [comment]
I thought that I applied both changes but actually only applied the 2. (changing `import` to `require`).

Only applying the change to the declaration file leads to `TS2307:Cannot find module '../aurelia.json'.` for the line `import * as project from '../aurelia.json';`.

Hope this helps

**Comment by sbolel** [comment]
@suamikim thanks for the response. @EisenbergEffect I'll look into this a bit more, then PR.

**Comment by sbolel** [comment]
This is not an issue on macOS. Even though my editor complains about not finding `../aurelia.json`, the build process works fine. I don't need the custom type definition either.

I unfortunately do not have a copy of windows, so I'm not sure how to go about trying to dive further into this. It might be that the fixes in #499 may resolve this?

As for the editor/IDE warning, we could add `json.d.ts` to the `custom_typings` folder if necessary.

**Comment by eric-914** [comment]
My 2¢:

Using `au new` to build my project, the following line compiles from the command line, but errors in Visual Studio:

`import * as project from '../aurelia.json';`

but is ok***, if I change it to:

`const project = require('../aurelia.json');`

_***Well, almost ok -- "cannot resolve 'require'", but I suspect that is ReSharper at the moment, as it builds fine._
